src/transformStateWithClones.js

```diff
@@ -8,6 +8,41 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+
+  let stateful = { ...state };
+  let copy = { ...state };
+  const copyNew = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copy, action.extraData);
+        result.push(copy);
+        stateful = { ...copy };
```

Comment:
No need to push in each case, you could do it once in the end of iteration, after `switch case`.